src/main/java/io/cdap/plugin/sfmc/source/MarketingCloudClient.java

```diff
@@ -204,9 +212,34 @@ private ETExpression buildDataExtensionExpression(String dataExtensionKey) {
 
 
   private List<MarketingCloudColumn> fetchObjectFields(Class<?> clazz) {
-    return Arrays.stream(clazz.getDeclaredFields())
-      .map(o -> new MarketingCloudColumn(o.getName(), o.getType().getSimpleName().toUpperCase()))
-      .collect(Collectors.toList());
+    SoapObject internalTypeAnnotation
+            = clazz.getAnnotation(SoapObject.class);
+    assert internalTypeAnnotation != null;
```

Comment:
Since I am not sure what the workflow is, I wonder if throwing an error saying something is null is actually helping the user. Does showing this exception to the user lead to the user correcting those fields and retrying the workflow? If not, we shouldn't throw this error.
Same for the next assert as well